walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,65 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      List<String> meetingAttendees = request.getAttendees().stream().collect(Collectors.toList());
+      long meetingDuration = request.getDuration();
+
+      // Find conflicting time ranges
+      List<TimeRange> conflictingTimeRanges =  getConflictingTimeRanges(events, meetingAttendees);
+      
+      // Find meeting time ranges
+      List<TimeRange> meetingTimeRanges = getMeetingTimeRanges(conflictingTimeRanges, meetingDuration);
+
+      return meetingTimeRanges;
+  }
+
+  // Function to return list of conflicting time slots
+  private List<TimeRange> getConflictingTimeRanges(Collection<Event> events, List<String> meetingAttendees){
+
+      List<TimeRange> conflictingTimeRanges = new ArrayList<>();
+      for(Event event:events){
+          Set<String> eventAttendees = event.getAttendees();
+          if ((eventAttendees.stream().filter(meetingAttendees::contains).collect(Collectors.toList())).size() > 0){
+              conflictingTimeRanges.add(event.getWhen());
+          }
+      }
```

Comment:
this entire block can be converted into a lambda

Reply:
Updated the code